StardewArchipelago/GameModifications/Modded/JunimoShopStockModifier.cs

```diff
@@ -0,0 +1,354 @@
+            if (soldItemValue > requestedItemValue && soldItemValue % requestedItemValue == 0)
+            {
+                exchangeRate = new int[2] { 1, soldItemValue / requestedItemValue };
+                return true;
+            }
+            if (soldItemValue <= requestedItemValue && requestedItemValue % soldItemValue == 0)
+            {
+                exchangeRate = new int[2] { requestedItemValue / soldItemValue, 1 };
+                return true;
+            }
+
+            return false;
+        }
+
+        private int StockBasedOnApplesFriendship()
+        {
+            var applesHearts = 0;
+            if (Game1.player.friendshipData.ContainsKey("Apples"))
+            {
+                applesHearts = Game1.player.friendshipData["Apples"].Points / 500;
+            }
+            return applesHearts + 1;
+        }
+
+        private int[] GiveApplesFriendshipDiscount(int soldItemCount, int requestedItemCount)
+        {
+            var applesHearts = 0;
+            if (Game1.player.friendshipData.ContainsKey("Apples"))
+            {
+                applesHearts = Game1.player.friendshipData["Apples"].Points / 250; // Get discount from being friends with Apples
+            }
+            if (requestedItemCount == 1)
+            {
+                soldItemCount = (int)(soldItemCount * (1 + applesHearts * 0.05f));
+            }
+            else
+            {
+                requestedItemCount = (int)Math.Max(1, requestedItemCount * (1 - applesHearts * 0.05f));
+            }
+            return new int[2] { soldItemCount, requestedItemCount };
+        }
+
+        private int[] MakeMinimalCountBelowGivenCount(int soldItemCount, int requestedItemCount, int givenCount)
+        {
+            if (Math.Min(soldItemCount, requestedItemCount) > givenCount)
+            {
+                var closestCount = (int)Math.Pow(givenCount, (int)(Math.Log10(Math.Min(soldItemCount, requestedItemCount)) / Math.Log10(givenCount)));
+                soldItemCount /= closestCount;
+                requestedItemCount /= closestCount;
+                var greatestCommonDivisor = GreatestCommonDivisor(soldItemCount, requestedItemCount); // Due to the rounding we may find the two aren't relatively prime anymore
+                soldItemCount /= greatestCommonDivisor;
+                requestedItemCount /= greatestCommonDivisor;
+            }
+            return new int[2] { soldItemCount, requestedItemCount };
+        }
+
+        private ShopItemData CreateJunimoSeedItem(string qualifiedId, string[] season = null)
+        {
+            var seedItemName = _stardewItemManager.GetItemByQualifiedId(qualifiedId).Name;
+            var condition = season is not null ? $"{GameStateConditionProvider.CreateSeasonsCondition(season)}, {GameStateConditionProvider.CreateHasReceivedItemCondition(seedItemName)}, RANDOM 0.4 @addDailyLuck" : $"{GameStateConditionProvider.CreateHasReceivedItemCondition(seedItemName)}, RANDOM 0.4 @addDailyLuck";
```

Comment:
Same comment as before, the part of the query that is the same should be built outside of the ternary